ext/src/ruby_api/wasi_ctx_builder.rs

```diff
@@ -222,20 +225,32 @@ impl WasiCtxBuilder {
         rb_self
     }
 
+    /// @yard
+    /// Make the context deterministic. See See https://github.com/Shopify/deterministic-wasi-ctx for more details
+    /// @def add_determinism()
+    /// @return [WasiCtxBuilder] +self+
+    pub fn add_determinism(rb_self: RbSelf) -> RbSelf {
```

Comment:
Note that we still need to call `replace_scheduling_functions` (and the preview 0 equivalent) with `deterministic_wasi_ctx` passing the linker as an argument if there's a desire to prevent modules from being able to sleep.